python/ray/dag/dag_node_operation.py

```diff
@@ -0,0 +1,337 @@
+from functools import total_ordering
+from enum import Enum
+from typing import Set, Tuple, List, Dict, Optional
+import ray
+import heapq
+from collections import defaultdict
+
+
+class _DAGNodeOperationType(Enum):
+    """
+    There are three types of operations that a DAG node can perform:
+    1. READ: Read from an input channel.
+    2. COMPUTE: Execute the method corresponding to the node.
+    3. WRITE: Write to an output channel.
+    """
+
+    READ = "READ"
+    COMPUTE = "COMPUTE"
+    WRITE = "WRITE"
+
+
+class _DAGNodeOperation:
+    def __init__(
+        self,
+        local_idx: int,
+        operation_type: _DAGNodeOperationType,
+    ):
+        """
+        Args:
+            local_idx: The index of the task that this operation belongs to
+                in the actor's ExecutableTask list. The index is not the same
+                as bind_index, but there are positive correlations between the two.
```

Comment:
```suggestion
                in the actor's ExecutableTask list. The index is not the same
                as bind_index because there may be more tasks bound to an actor than tasks that appear in the current compiled DAG.
```